client/src/redux/actions/debt.js

```diff
@@ -0,0 +1,46 @@
+import { FETCH_DEBTS_SUCCESS, SAVE_DEBT_SUCCESS } from "./types";
+
+import getDebts, { saveDebtApi } from "../../mockApi/api/debtApi";
+
+// synchronous actions
+const updateDebtSuccess = (id = "", updates = {}) => ({
+  type: SAVE_DEBT_SUCCESS,
+  id,
+  updates
+});
+
+const fetchDebtSuccess = (debt = []) => ({
+  type: FETCH_DEBTS_SUCCESS,
+  debt
+});
+
+// Asynchronous actions -> Thunks
+const fetchDebts = () => {
+  return async dispatch => {
+    try {
+      const response = await getDebts();
+      if (response) {
+        return dispatch(
+          fetchDebtSuccess(
+            response.map(debt => ({ ...debt, value: parseFloat(debt.value) })) //Parse to double
+          )
+        );
+      }
+    } catch (error) {
+      throw error;
```

Comment:
Throwing an error is fine for now, but i'd rather not crash the whole app. Once snackbars are set up, I would like to just notify the user and do any cleanup logic if needed.